src/sentry/static/sentry/app/views/organizationGroupDetails/groupDetails.tsx

```diff
@@ -87,6 +110,128 @@ class GroupDetails extends React.Component<Props, State> {
     return `/issues/${this.props.params.groupId}/`;
   }
 
+  async getEvent(group: Group) {
+    const {params, environments, api, organization} = this.props;
+    const orgSlug = organization.slug;
+    const groupId = group.id;
+    const projSlug = group.project.slug;
+    const eventId = params?.eventId || 'latest';
+
+    try {
+      const event = await fetchGroupEventAndMarkSeen(
+        api,
+        orgSlug,
+        projSlug,
+        groupId,
+        eventId,
+        environments
+      );
+      this.setState({event, loading: false, error: false, errorType: null});
+    } catch (err) {
+      // This is an expected error, capture to Sentry so that it is not considered as an unhandled error
+      Sentry.captureException(err);
+      this.setState({error: true, errorType: null, loading: false});
+    }
+  }
+
+  getRouteInfo() {
+    const {routes} = this.props;
+    const route = getRouteStringFromRoutes(routes);
+
+    switch (route) {
```

Comment:
This is a lot, and can be a bit brittle... is it possible to add these as props onto the route themselves in `app/routes`?

Reply:
yeah its a good idea... I will update it